eventmesh/ApiServerEventMesh.js

```diff
@@ -0,0 +1,411 @@
+'use strict';
+
+const _ = require('lodash');
+const Promise = require('bluebird');
+const assert = require('assert');
+const config = require('../common/config');
+const logger = require('../common/logger');
+const CONST = require('../common/constants');
+const EventMeshServer = require('./EventMeshServer');
+const kc = require('kubernetes-client');
+const JSONStream = require('json-stream');
+const errors = require('../common/errors');
+const BadRequest = errors.BadRequest;
+const NotFound = errors.NotFound;
+const Conflict = errors.Conflict;
+const InternalServerError = errors.InternalServerError;
+
+const apiserver = new kc.Client({
+  config: {
+    url: `https://${config.internal.ip}:${CONST.APISERVER.PORT}`,
+    insecureSkipTlsVerify: true
+  },
+  version: CONST.APISERVER.VERSION
+});
+
+function buildErrors(err) {
+  let throwErr;
+  switch (err.code) {
+  case CONST.HTTP_STATUS_CODE.BAD_REQUEST:
+    throwErr = new BadRequest(err.message);
+    break;
+  case CONST.HTTP_STATUS_CODE.NOT_FOUND:
+    throwErr = new NotFound(err.message);
+    break;
+  case CONST.HTTP_STATUS_CODE.CONFLICT:
+    throwErr = new Conflict(err.message);
+    break;
+  default:
+    throwErr = new InternalServerError(err.message);
+    break;
+  }
+  throw throwErr;
+}
+
+class ApiServerEventMesh extends EventMeshServer {
+  /**
+   * Register watcher for (resourceName , resourceType)
+   * @param {string} resourceName
```

Comment:
add descriptions for the arguments for readability

Reply:
Done